### DISCUSSION TITLE
adm-zip needs extra config to work because of dynamic import for electron

### DISCUSSION THREAD

**Discussion by antoinepairet** [web-infra-dev/rspack#3038] [Core]
When one tries to bundle all dependencies (aka `externals` is an empty array), `rspack` does not work with `adm-zip` out of the box. `rspack` tries to resolve `original-fs` because there is a dynamic import to cover the `electron` runtime.

The issue comes from this file: `https://github.com/cthackers/adm-zip/blob/master/util/fileSystem.js`:
````js
exports.require = function () {
    if (typeof process === "object" && process.versions && process.versions["electron"]) {
        try {
            const originalFs = require("original-fs");
            if (Object.keys(originalFs).length > 0) {
                return originalFs;
            }
        } catch (e) {}
    }
    return require("fs");
};
````

There are two workarounds:
- installing `original-fs` as a dependency
- putting `original-fs` in the external array in such a way that `rspack` ignores it

Not sure if there should be an action point for `rspack` at this stage, maybe mentioning such a case in the docs?
I assume other packages might use the same technique as the one `adm-zip` is using.

**Comment by hardfist** [comment]
https://github.com/web-infra-dev/rspack/issues/3042 once we support this is should be fixed

**Comment by hardfist** [comment]
@antoinepairet  since #3042 is solved, can you try it on your project

**Reply by antoinepairet** [reply]
Works perfectly. Thank you 🙌🏻
